tools/inspect/Inspect.cpp

```diff
@@ -213,47 +217,144 @@ int parse_args(int argc, ACE_TCHAR* argv[])
   return 0;
 }
 
-class RecorderListenerImpl : public RecorderListener {
-public:
-  explicit RecorderListenerImpl(DDS::GuardCondition_var gc)
-    : sample_count_(0)
-    , gc_(gc)
+bool check_rc(DDS::ReturnCode_t rc, const std::string& what)
+{
+  if (rc != DDS::RETCODE_OK) {
+    ACE_ERROR((LM_ERROR, "ERROR: %C: %C\n", what.c_str(), retcode_to_string(rc)));
+    return false;
+  }
+  return true;
+}
+
+struct Topic : public virtual LocalObject<DDS::DataReaderListener> {
+  std::string name;
+  std::string type_name;
+  std::string path;
+  DDS::DynamicType_var type;
+  DDS::TypeSupport_var ts;
+  DDS::Topic_var topic;
+  DDS::DataReader_var reader;
+  DDS::GuardCondition_var gc;
+
+  Topic(const std::string& a_name, const std::string& a_type_name, DDS::GuardCondition_var a_gc)
+  : name(a_name)
+  , type_name(a_type_name)
+  , gc(a_gc)
+  {
+  }
+
+  bool matches_topic_queries()
```

Comment:
```suggestion
  bool matches_topic_queries() const
```
Not that it will really matter...